File: src/main/java/io/github/chaosunity/forgelin/preloader/ForgelinPlugin.java
package io.github.chaosunity.forgelin.preloader;

import net.minecraftforge.fml.relauncher.IFMLLoadingPlugin;

import java.util.Map;

public class ForgelinPlugin implements IFMLLoadingPlugin {
    @Override
    public String[] getASMTransformerClass() {
        return new String[0];
    }

    @Override
    public String getModContainerClass() {
        return null;
    }

    @Override
    public String getSetupClass() {
        return "io.github.chaos.forgelin.preloader.ForgelinSetup";
    }

    @Override
    public void injectData(Map<String, Object> data) { }

    @Override
    public String getAccessTransformerClass() {
        return null;
    }
}

File: src/main/kotlin/io/github/chaosunity/forgelin/preloader/ForgelinPlugin.kt
package io.github.chaosunity.forgelin.preloader

import com.ibm.icu.impl.PluralRulesLoader.loader
import net.minecraftforge.common.ForgeVersion
import net.minecraftforge.fml.relauncher.IFMLLoadingPlugin
import org.apache.logging.log4j.LogManager

@IFMLLoadingPlugin.Name("Forgelin-Continuous")
@IFMLLoadingPlugin.MCVersion(ForgeVersion.mcVersion)
class ForgelinPlugin: IFMLLoadingPlugin {
    override fun getASMTransformerClass(): Array<String> {
        return arrayOf()
    }

    override fun getModContainerClass(): String? {
        return null
    }

    override fun getSetupClass(): String {
        return "io.github.chaosunity.forgelin.preloader.ForgelinSetup"
    }

    override fun injectData(data: MutableMap<String, Any>?) {}

    override fun getAccessTransformerClass(): String {
        return "io.github.chaosunity.forgelin.transformer.ForgelinTransformer"
    }
}